ssl/s3_enc.c

```diff
@@ -299,6 +299,7 @@ int ssl3_setup_key_block(SSL *s)
     s->s3->tmp.key_block = p;
 
     ret = ssl3_generate_key_block(s, p, num);
+    if (ret==0) goto err;
```

Comment:
Not once its assigned to `s->s3->tmp.key_block`. From that point ownership is transferred to the SSL object.